diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0567ae7..60f33d1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,6 +10,12 @@ if(NOT CYGWIN AND NOT MSYS AND NOT ${CMAKE_SYSTEM_NAME} STREQUAL QNX)
   set(CMAKE_CXX_EXTENSIONS OFF)
 endif()
 
+if(MSVC)
+  # Always apply postfix for Debug configs in MSVC
+  set(CMAKE_DEBUG_POSTFIX d)
+  message(STATUS "MSVC detected: Debug postfix enabled ('d')")
+endif()
+
 enable_testing()
 
 include(CMakeDependentOption)
